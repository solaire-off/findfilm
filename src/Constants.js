export const API_ROOT = "http://localhost:4000";
export const SITE_NAME = ["netflix", "roulette"];
export const GENRES_LIST = ["Documentary", "Comedy", "Horror", "Crime"];
export const DEFAULT_SORT_FIELD = "vote_count";
export const SORT_TYPES = [
  {
    name: "popularity",
    value: DEFAULT_SORT_FIELD,
  },
  {
    name: "release date",
    value: "release_date",
  },
  {
    name: "rating",
    value: "vote_average",
  },
  {
    name: "name",
    value: "title",
  },
];
export const FETCH_FILMS_COUNT = 18;
export const DEFAULT_GENRE_ANY = "All";
export const EXAMPLE_FILM = {
  title: "La La Land",
  tagline: "Here's to the fools who dream.",
  vote_average: 7.9,
  vote_count: 6782,
  release_date: "2016-12-29",
  poster_path:
    "https://image.tmdb.org/t/p/w500/ylXCdC106IKiarftHkcacasaAcb.jpg",
  overview:
    "Mia, an aspiring actress, serves lattes to movie stars in between auditions and Sebastian, a jazz musician, scrapes by playing cocktail party gigs in dingy bars, but as success mounts they are faced with decisions that begin to fray the fragile fabric of their love affair, and the dreams they worked so hard to maintain in each other threaten to rip them apart.",
  budget: 30000000,
  revenue: 445435700,
  runtime: 128,
  genres: ["Comedy", "Drama", "Romance"],
  id: 313369,
};
export const EXAMPLE_FILM_LIST = [
  {
    id: 337167,
    title: "Fifty Shades Freed",
    tagline: "Don't miss the climax",
    vote_average: 6.1,
    vote_count: 1195,
    release_date: "2018-02-07",
    poster_path:
      "https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg",
    overview:
      "Believing they have left behind shadowy figures from their past, newlyweds Christian and Ana fully embrace an inextricable connection and shared life of luxury. But just as she steps into her role as Mrs. Grey and he relaxes into an unfamiliar stability, new threats could jeopardize their happy ending before it even begins.",
    budget: 55000000,
    revenue: 136906000,
    genres: ["Drama", "Romance"],
    runtime: 106,
  },
  {
    id: 269149,
    title: "Zootopia",
    tagline: "Welcome to the urban jungle.",
    vote_average: 7.7,
    vote_count: 6795,
    release_date: "2016-02-11",
    poster_path:
      "https://image.tmdb.org/t/p/w500/sM33SANp9z6rXW8Itn7NnG1GOEs.jpg",
    overview:
      "Determined to prove herself, Officer Judy Hopps, the first bunny on Zootopia's police force, jumps at the chance to crack her first case - even if it means partnering with scam-artist fox Nick Wilde to solve the mystery.",
    budget: 150000000,
    revenue: 1023784195,
    genres: ["Animation", "Adventure", "Family", "Comedy"],
    runtime: 108,
  },
  {
    id: 181808,
    title: "Star Wars: The Last Jedi",
    tagline: "The Saga Continues",
    vote_average: 7.1,
    vote_count: 4732,
    release_date: "2017-12-13",
    poster_path:
      "https://image.tmdb.org/t/p/w500/kOVEVeg59E0wsnXmF9nrh6OmWII.jpg",
    overview:
      "Rey develops her newly discovered abilities with the guidance of Luke Skywalker, who is unsettled by the strength of her powers. Meanwhile, the Resistance prepares to do battle with the First Order.",
    budget: 200000000,
    revenue: 1325937250,
    genres: ["Fantasy", "Adventure", "Science Fiction"],
    runtime: 152,
  },
];
