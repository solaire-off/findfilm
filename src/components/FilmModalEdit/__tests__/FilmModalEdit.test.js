import React from "react";
import { render, act } from "@testing-library/react";
import { FilmModalEdit } from "..";
import { EXAMPLE_FILM } from "../../../Constants";
import { MemoryRouter } from "react-router-dom";

global.fetch = jest.fn(() =>
  Promise.resolve({
    json: () => Promise.resolve(EXAMPLE_FILM),
  })
);

window.scroll = () => {}; // provide an empty implementation for window.scroll

it("Current render when 'add' mode", async () => {
  await act(async () => {
    render(
      <MemoryRouter initialentries={[{ pathname: "/search" }]}>
        <FilmModalEdit modalTitle="Add film" sendFilmData={() => {}} />
      </MemoryRouter>
    );
  });
  expect(document.body).toMatchSnapshot();
});

it("Current render 'edit' mode", async () => {
  await act(async () => {
    render(
      <MemoryRouter initialentries={[{ pathname: "/search" }]}>
        <FilmModalEdit
          modalTitle="Edit modal"
          id={EXAMPLE_FILM.id}
          sendFilmData={() => {}}
        />
      </MemoryRouter>
    );
  });
  expect(document.body).toMatchSnapshot();
});
