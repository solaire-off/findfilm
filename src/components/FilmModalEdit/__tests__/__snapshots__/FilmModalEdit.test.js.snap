// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Current render when 'add' mode 1`] = `
<body>
  <div>
    <form
      action="."
      method="GET"
    >
      <div
        class="modal__header"
      >
        <p
          class="modal__title"
        >
          Add film
        </p>
        <button
          aria-label="Close"
          class="modal__close"
          type="button"
        />
      </div>
      <div
        class="modal__body"
      >
        <div
          class="film-modal__row film-modal__group"
        >
          <label
            class="label"
          >
            <span
              class="label__caption"
            >
              TITLE
            </span>
            <input
              class="form-control  "
              placeholder="Movie title"
              type="text"
              value=""
            />
          </label>
          <label
            class="label"
          >
            <span
              class="label__caption"
            >
              RELEASE DATE
            </span>
            <div
              class="react-datepicker-wrapper"
            >
              <div
                class="react-datepicker__input-container"
              >
                <input
                  class=""
                  placeholder="Select Release Date"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </label>
          <label
            class="label"
          >
            <span
              class="label__caption"
            >
              MOVIE URL
            </span>
            <input
              class="form-control  "
              placeholder="https://"
              type="text"
              value=""
            />
          </label>
          <label
            class="label"
          >
            <span
              class="label__caption"
            >
              RATING
            </span>
            <input
              class="form-control  "
              placeholder="from 1 to 10"
              type="text"
              value=""
            />
          </label>
          <label
            class="label"
          >
            <span
              class="label__caption"
            >
              GENRE
            </span>
            <div
              class="react-select css-b62m3t-container"
            >
              <span
                class="css-1f43avz-a11yText-A11yText"
                id="react-select-2-live-region"
              />
              <span
                aria-atomic="false"
                aria-live="polite"
                aria-relevant="additions text"
                class="css-1f43avz-a11yText-A11yText"
              />
              <div
                class="react-select__control css-gfnnwk-control"
              >
                <div
                  class="react-select__value-container react-select__value-container--is-multi css-319lph-ValueContainer"
                >
                  <div
                    class="react-select__placeholder css-14el2xx-placeholder"
                    id="react-select-2-placeholder"
                  >
                    Select Genres
                  </div>
                  <div
                    class="react-select__input-container css-14dclt2-Input"
                    data-value=""
                  >
                    <input
                      aria-autocomplete="list"
                      aria-controls="react-select-2-listbox"
                      aria-describedby="react-select-2-placeholder"
                      aria-expanded="false"
                      aria-haspopup="true"
                      aria-owns="react-select-2-listbox"
                      autocapitalize="none"
                      autocomplete="off"
                      autocorrect="off"
                      class="react-select__input"
                      id="react-select-2-input"
                      role="combobox"
                      spellcheck="false"
                      style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                      tabindex="0"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="react-select__indicators css-1hb7zxy-IndicatorsContainer"
                >
                  <span
                    class="react-select__indicator-separator css-1okebmr-indicatorSeparator"
                  />
                  <div
                    aria-hidden="true"
                    class="react-select__indicator react-select__dropdown-indicator css-tlfecz-indicatorContainer"
                  >
                    <svg
                      aria-hidden="true"
                      class="css-tj5bde-Svg"
                      focusable="false"
                      height="20"
                      viewBox="0 0 20 20"
                      width="20"
                    >
                      <path
                        d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </div>
          </label>
          <label
            class="label"
          >
            <span
              class="label__caption"
            >
              RUNTIME
            </span>
            <input
              class="form-control  "
              placeholder="Minutes"
              type="text"
              value=""
            />
          </label>
        </div>
        <div
          class="film-modal__group"
        >
          <label
            class="label"
          >
            <span
              class="label__caption"
            >
              OVERVIEW
            </span>
            <textarea
              class="form-control   form-control--textarea"
              placeholder="Movie description"
            />
          </label>
        </div>
      </div>
      <div
        class="modal__footer modal__footer--mt"
      >
        <button
          class="btn btn--outline-danger btn--lg modal__btn font-weight-medium"
          type="button"
        >
          Reset
        </button>
        <button
          class="btn btn--primary btn--lg modal__btn font-weight-medium"
          type="submit"
        >
          Submit
        </button>
      </div>
    </form>
  </div>
</body>
`;

exports[`Current render when 'edit' mode 1`] = `
<body>
  <div>
    <form
      action="."
      method="GET"
    >
      <div
        class="modal__header"
      >
        <p
          class="modal__title"
        >
          Edit modal
        </p>
        <button
          aria-label="Close"
          class="modal__close"
          type="button"
        />
      </div>
      <div
        class="modal__body"
      >
        <div
          class="film-modal__row film-modal__group"
        >
          <label
            class="label"
          >
            <span
              class="label__caption"
            >
              TITLE
            </span>
            <input
              class="form-control  "
              placeholder="Movie title"
              type="text"
              value="La La Land"
            />
          </label>
          <label
            class="label"
          >
            <span
              class="label__caption"
            >
              RELEASE DATE
            </span>
            <div
              class="react-datepicker-wrapper"
            >
              <div
                class="react-datepicker__input-container"
              >
                <input
                  class=""
                  placeholder="Select Release Date"
                  type="text"
                  value="2016-12-29"
                />
              </div>
            </div>
          </label>
          <label
            class="label"
          >
            <span
              class="label__caption"
            >
              MOVIE URL
            </span>
            <input
              class="form-control  "
              placeholder="https://"
              type="text"
              value="https://image.tmdb.org/t/p/w500/ylXCdC106IKiarftHkcacasaAcb.jpg"
            />
          </label>
          <label
            class="label"
          >
            <span
              class="label__caption"
            >
              RATING
            </span>
            <input
              class="form-control  "
              placeholder="from 1 to 10"
              type="text"
              value="7.9"
            />
          </label>
          <label
            class="label"
          >
            <span
              class="label__caption"
            >
              GENRE
            </span>
            <div
              class="react-select css-b62m3t-container"
            >
              <span
                class="css-1f43avz-a11yText-A11yText"
                id="react-select-3-live-region"
              />
              <span
                aria-atomic="false"
                aria-live="polite"
                aria-relevant="additions text"
                class="css-1f43avz-a11yText-A11yText"
              />
              <div
                class="react-select__control css-gfnnwk-control"
              >
                <div
                  class="react-select__value-container react-select__value-container--is-multi react-select__value-container--has-value css-g1d714-ValueContainer"
                >
                  <div
                    class="react-select__placeholder css-14el2xx-placeholder"
                    id="react-select-3-placeholder"
                  >
                    Select Genres
                  </div>
                  <div
                    class="react-select__input-container css-14dclt2-Input"
                    data-value=""
                  >
                    <input
                      aria-autocomplete="list"
                      aria-controls="react-select-3-listbox"
                      aria-expanded="false"
                      aria-haspopup="true"
                      aria-owns="react-select-3-listbox"
                      autocapitalize="none"
                      autocomplete="off"
                      autocorrect="off"
                      class="react-select__input"
                      id="react-select-3-input"
                      role="combobox"
                      spellcheck="false"
                      style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                      tabindex="0"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="react-select__indicators css-1hb7zxy-IndicatorsContainer"
                >
                  <div
                    aria-hidden="true"
                    class="react-select__indicator react-select__clear-indicator css-tlfecz-indicatorContainer"
                  >
                    <svg
                      aria-hidden="true"
                      class="css-tj5bde-Svg"
                      focusable="false"
                      height="20"
                      viewBox="0 0 20 20"
                      width="20"
                    >
                      <path
                        d="M14.348 14.849c-0.469 0.469-1.229 0.469-1.697 0l-2.651-3.030-2.651 3.029c-0.469 0.469-1.229 0.469-1.697 0-0.469-0.469-0.469-1.229 0-1.697l2.758-3.15-2.759-3.152c-0.469-0.469-0.469-1.228 0-1.697s1.228-0.469 1.697 0l2.652 3.031 2.651-3.031c0.469-0.469 1.228-0.469 1.697 0s0.469 1.229 0 1.697l-2.758 3.152 2.758 3.15c0.469 0.469 0.469 1.229 0 1.698z"
                      />
                    </svg>
                  </div>
                  <span
                    class="react-select__indicator-separator css-1okebmr-indicatorSeparator"
                  />
                  <div
                    aria-hidden="true"
                    class="react-select__indicator react-select__dropdown-indicator css-tlfecz-indicatorContainer"
                  >
                    <svg
                      aria-hidden="true"
                      class="css-tj5bde-Svg"
                      focusable="false"
                      height="20"
                      viewBox="0 0 20 20"
                      width="20"
                    >
                      <path
                        d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </div>
          </label>
          <label
            class="label"
          >
            <span
              class="label__caption"
            >
              RUNTIME
            </span>
            <input
              class="form-control  "
              placeholder="Minutes"
              type="text"
              value="128"
            />
          </label>
        </div>
        <div
          class="film-modal__group"
        >
          <label
            class="label"
          >
            <span
              class="label__caption"
            >
              OVERVIEW
            </span>
            <textarea
              class="form-control   form-control--textarea"
              placeholder="Movie description"
            >
              Mia, an aspiring actress, serves lattes to movie stars in between auditions and Sebastian, a jazz musician, scrapes by playing cocktail party gigs in dingy bars, but as success mounts they are faced with decisions that begin to fray the fragile fabric of their love affair, and the dreams they worked so hard to maintain in each other threaten to rip them apart.
            </textarea>
          </label>
        </div>
      </div>
      <div
        class="modal__footer modal__footer--mt"
      >
        <button
          class="btn btn--outline-danger btn--lg modal__btn font-weight-medium"
          type="button"
        >
          Reset
        </button>
        <button
          class="btn btn--primary btn--lg modal__btn font-weight-medium"
          type="submit"
        >
          Submit
        </button>
      </div>
    </form>
  </div>
</body>
`;
